cmd/compute-domain-controller/resourceclaimtemplate.go

```diff
@@ -23,14 +23,14 @@ import (
 	"sync"
 	"text/template"
 
-	resourceapi "k8s.io/api/resource/v1beta1"
+	resourceapi "k8s.io/api/resource/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/yaml"
-	"k8s.io/client-go/informers"
+	"k8s.io/apimachinery/pkg/watch"
```

Comment:
what's the background for this change?

Reply:
We can't use a factory anymore, because we need to intercept the informers List and Watch calls and force them to use Patrick's custom `Resource` client library rather than the standard `Core` client library.

Patrick's client knows how to detect the latest version of the `Resource` API available in the cluster is and use that rather than being forced to specify a specific one.

With this in place we are able to manipulate all data structures using the v1 API and trust that the client library translates this to the actual API supported by the cluster (v1, v1beta2, or v1beta1).

The new `watch` import is just an artifact of needing to create the informer directly instead of relying on the factory to do it for us.